app/controllers/posts_controller.rb

```diff
@@ -1,21 +1,30 @@
 class PostsController < ApplicationController
   def index
-    # @user = User.find_by(id: params[:user_id])
+    @user = User.find_by(id: params[:user_id])
     # @posts = @user.posts
-    @posts = Post.where(user_id: post_params[:user_id])
+    # @posts = Post.where(user_id: post_params[:user_id])
+    @posts = Post.where(user_id: params[:user_id])
+    puts @posts[0].comments
   end
 
   def show
-    # @user = User.find_by(id: params[:user_id])
+    @user = User.find_by(id: params[:user_id])
     # @posts = @user.posts
     # @post = @posts.find_by(id: params[:post_id])
-    @post = Post.find_by(user_id: post_params[:user_id], id: params[:post_id])
+    @post = Post.find_by(user_id: params[:user_id], id: params[:id])
+
+    # @post = Post.find_by(user_id: params[:user_id],id: params[:post_id])
     @comments = @post.comments
-    @user = User.find(@post.user_id)
+
+    puts 'My posts:'
+    puts @post
+    puts '....'
+
+    # @user = User.find_by(user_id: params[:user_id])
```

Comment:
- [ ] Let's do a quick cleanup in our code, we have a lot of commented code which makes our code look so untidy.
- [ ] One of the project requirements is to fix the `N + 1` query problem. To solve this issue, we are expected to query posts for a user together with their comments in our index action. There are a lot of Ruby built-in methods to solve issues. We can do something like this: `@user.posts.includes(:comments)`. Notice that I have used the `includes` method to join two tables (posts and comments)